json_schema/core/file/file_core.json

```diff
@@ -1,76 +1,76 @@
             "example": "fastq.gz; tif"
-        }, 
+        },
         "content_description": {
-            "description": "General description of the contents of the file.", 
-            "type": "array", 
-            "user_friendly": "Content description", 
+            "description": "General description of the contents of the file.",
+            "type": "array",
+            "user_friendly": "Content description",
             "items": {
-                "type": "object", 
-                "$ref": "https://raw.githubusercontent.com/morphic-bio/metadata-schema/main/json_schema/module/ontology/1.0.0/file_content_ontology"
+                "type": "object",
+                "$ref": "module/ontology/file_content_ontology.json"
             }
-        }, 
+        },
         "checksum": {
-            "description": "MD5 checksum of the file.", 
-            "type": "string", 
-            "user_friendly": "Checksum", 
+            "description": "MD5 checksum of the file.",
+            "type": "string",
+            "user_friendly": "Checksum",
             "example": "e09a986c2e630130b1849d4bf9a94c06"
-        }, 
+        },
         "file_source": {
-            "description": "The source of the file. This is typically an organisation, repository, person or dedicated process.", 
-            "type": "string", 
+            "description": "The source of the file. This is typically an organisation, repository, person or dedicated process.",
+            "type": "string",
             "enum": [
-                "DCP/2 Analysis", 
-                "Contributor", 
-                "ArrayExpress", 
-                "HCA Release", 
-                "GEO", 
-                "SCEA", 
-                "SCP", 
-                "DCP/1 Matrix Service", 
-                "LungMAP", 
-                "Zenodo", 
-                "Publication", 
+                "DCP/2 Analysis",
+                "Contributor",
+                "ArrayExpress",
+                "HCA Release",
+                "GEO",
+                "SCEA",
+                "SCP",
+                "DCP/1 Matrix Service",
+                "LungMAP",
+                "Zenodo",
+                "Publication",
                 "DCP/2 Ingest"
-            ], 
-            "user_friendly": "File source", 
-            "guidelines": "Should be one of: DCP/2 Analysis, Contributor, ArrayExpress, HCA Release, GEO, SCEA, SCP, DCP/1 Matrix Service, LungMAP, Zenodo, Publication, DCP/2 Ingest", 
+            ],
+            "user_friendly": "File source",
+            "guidelines": "Should be one of: DCP/2 Analysis, Contributor, ArrayExpress, HCA Release, GEO, SCEA, SCP, DCP/1 Matrix Service, LungMAP, Zenodo, Publication, DCP/2 Ingest",
             "example": "GEO; Contributor"
         }
-    }, 
+    },
     "$id": "https://raw.githubusercontent.com/morphic-bio/metadata-schema/main/json_schema/core/file/1.0.0/file_core"
```

Comment:
If we're pushing to a bucket, should we delete the $id and let the script that pushes create that value?

Reply:
I like this idea. Usually version of the $id is the only field that changes frequently, can we handle this including the $ref values? 
I also prefer `..../file_core/1.0.0` instead of `.../1.0.0/file_core`. This allows to use the schema without version URL part for referring to latest schema.

Reply:
The script pushes to the bucket. It gets as input the domains where it is published. It updates the `$id` as well as the `$ref`s. As far as I know refs need to be absolute urls so the document is standalone.